### PR TITLE
[Bug] [seatunnel-connector-spark-redis] Support redis authentication

### LINKED ISSUE #1135: [Bug] [seatunnel-connector-spark-redis] Solve user unauthorized problem and code optimization
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Solve user unauthorized problem and code optimization

### SeaTunnel Version

2.0.5

### SeaTunnel Config

```conf
-
```

### Running Command

```shell
-
```

### Error Exception

```log
Caused by: redis.clients.jedis.exceptions.JedisDataException: NOAUTH Authentication required.
	at redis.clients.jedis.Protocol.processError(Protocol.java:132)
	at redis.clients.jedis.Protocol.process(Protocol.java:166)
	at redis.clients.jedis.Protocol.read(Protocol.java:220)
	at redis.clients.jedis.Connection.readProtocolWithCheckingBroken(Connection.java:318)
	at redis.clients.jedis.Connection.getStatusCodeReply(Connection.java:236)
	at redis.clients.jedis.BinaryJedis.select(BinaryJedis.java:560)
	at redis.clients.jedis.JedisFactory.makeObject(JedisFactory.java:122)
	at org.apache.commons.pool2.impl.GenericObjectPool.create(GenericObjectPool.java:819)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:429)
	at org.apache.commons.pool2.impl.GenericObjectPool.borrowObject(GenericObjectPool.java:360)
	at redis.clients.jedis.util.Pool.getResource(Pool.java:50)
	... 20 more
```

### Flink or Spark Version

_No response_

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by wuchunfu**
…em and code optimization

## Purpose of this pull request

close #1135
## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [x] If any new Jar binary package adding in you PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/developement/NewLicenseGuide.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by RickyHuo** [comment]
PTAL @an-shi-chi-fan

File: seatunnel-connectors/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala [hunk 1]
```diff
@@ -16,113 +16,111 @@
...
+    config = config.withFallback(defaultConfig)
+  }
 
-    val conf = prepareEnv.getSparkSession.conf
-    conf.set("spark.redis.host", config.getString("redis_host"))
-    conf.set("spark.redis.port", config.getString("redis_port"))
-    if (config.hasPath("redis.auth")) {
-      conf.set("spark.redis.auth", "passwd")
+  def checkParam(checkArr: Array[String]): CheckResult = {
+    val notExistConfig: Array[String] = checkArr.filter(checkItem => !config.hasPath(checkItem))
```
Comment (simon824) [inline seatunnel-connectors/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala]: we can remove `checkParam()`

Comment (wuchunfu) [reply seatunnel-connectors/seatunnel-connector-spark-redis/src/main/scala/org/apache/seatunnel/spark/sink/Redis.scala]: > we can remove `checkParam()`

ok i'll fix that later.

**Review by CalvinKirs — APPROVED** [review]
Well done

**Comment by CalvinKirs** [comment]
It's better to update your title(support Redis authentication)

**Pull request merged**